sensing/altitude_estimation.hpp

```diff
@@ -74,33 +65,58 @@ typedef struct
 
 
 /**
- * \brief                       Initializes the altitude estimation structure
- *
- * \param   estimator           Pointer to data structure
- * \param   config              Pointer to configuration
- * \param   sonar               Pointer to the sonar
- * \param   barometer           Pointer to the barometer
- * \param   ahrs                Pointer to the ahrs
- * \param   altitude_estimated  Pointer to the estimated altitude
+ * \brief   Default configuration
+ * 
+ * \return  Configuration structure
  */
-void altitude_estimation_init(altitude_estimation_t* estimator,
-                              const altitude_estimation_conf_t* config,
-                              const sonar_t* sonar,
-                              const barometer_t* barometer,
-                              const ahrs_t* ahrs,
-                              altitude_t* altitude_estimated);
+static inline altitude_estimation_conf_t altitude_estimation_default_config(void);
 
 
 /**
- * \brief                   Main update function
- *
- * \param   estimator       Pointer to data structure
+ * \brief   Altitude estimator
  */
-void altitude_estimation_update(altitude_estimation_t* estimator);
+class Altitude_estimation: public Kalman<3,1,1>
+{
+public:
+    Altitude_estimation(Sonar& sonar, 
+                        Barometer& barometer, 
+                        ahrs_t& ahrs, 
+                        altitude_t& altitude, 
+                        altitude_estimation_conf_t config = altitude_estimation_default_config() );
+
+    /**
+     * \brief   Initialization
+     * 
+     * \return  Success
+     */
+    bool init(void);
+
+
+    /**
+     * \brief   Main update function
+     * 
+     * \return  Success
+     */
+    bool update(void);
+
 
+private:
+    const Sonar&        sonar_;           ///< Sonar, must be downward facing (input)
+    const Barometer&    barometer_;       ///< Barometer (input)
```

Comment:
You declare a reference to the barometer, but it seems that you are not updating the altitude estimation with it, isnt't it?